includes/class-taxjar-refund-record.php

```diff
@@ -27,9 +27,11 @@ public function should_sync() {
 			return false;
 		}
 
-		if ( empty( $this->order_completed_date ) ) {
-			$this->add_error( __( 'Refund failed validation - parent order has no completed date.', 'wc-taxjar' ) );
-			return false;
+		if ( apply_filters( 'taxjar_should_validate_order_completed_date', true ) ) {
```

Comment:
From the sounds of the issue and change, I would consider inverting this condition. This makes the validation of order completed date "opt-in" if the value is not set. 

If we instead add a filter for _bypassing_ the validation here it would feel safer since I would imagine we would want this validation in most cases. (e.g. - `bypass_order_completed_validation`).

Reply:
We talked about this in another place but putting here as well. The filter does default to true and will only return false if someone adds a custom filter to it in a plugin or in their funtions.php file. Also, the latest commit does make the intended functionality a little easier to read.